src/tests/end-to-end/common/page-controllers/page.ts

```diff
@@ -138,6 +138,15 @@ export class Page {
         });
     }
 
+    public async waitForAndClickSelector(selector: string): Promise<void> {
```

Comment:
Did you find that this was necessary for the test to work? Playwright documents that `Page.click` already begins by waiting for the selector in question, so I would expect that this is actually just doubling the timeout for the operation in question without making it more reliable.

Reply:
This was a misunderstanding on my part. I wasn't aware that Page.click waited for the selector, I've removed it!